genesyscloud/routing_queue_outbound_email_address/resource_genesyscloud_routing_queue_outbound_email_address.go

```diff
@@ -0,0 +1,136 @@
+
+	return updateRoutingQueueOutboundEmailAddress(ctx, d, meta)
+}
+
+func readRoutingQueueOutboundEmailAddress(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getRoutingQueueOutboundEmailAddressProxy(sdkConfig)
+	queueId := d.Id()
+
+	log.Printf("Reading outbound email address for queue %s", queueId)
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		queueEmailAddress, resp, getErr := proxy.getRoutingQueueOutboundEmailAddress(ctx, queueId)
+		if getErr != nil {
+			if util.IsStatus404(resp) {
+				return retry.RetryableError(fmt.Errorf("failed to read outbound email address for queue %s: %s", queueId, getErr))
+			}
+			return retry.NonRetryableError(fmt.Errorf("failed to read outbound email address for queue %s: %s", queueId, getErr))
+		}
+
+		cc := consistencyChecker.NewConsistencyCheck(ctx, d, meta, ResourceRoutingQueueOutboundEmailAddress())
+
+		_ = d.Set("queue_id", queueId)
+		resourcedata.SetNillableReference(d, "domain_id", queueEmailAddress.Domain)
+
+		// The route property is a **Inboundroute hence all the checks
+		if queueEmailAddress.Route != nil && *queueEmailAddress.Route != nil && (*queueEmailAddress.Route).Id != nil {
+			_ = d.Set("route_id", *(*queueEmailAddress.Route).Id)
+		}
+
+		log.Printf("Reading outbound email address for queue %s", queueId)
+		return cc.CheckState()
+	})
+}
+
+func updateRoutingQueueOutboundEmailAddress(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getRoutingQueueOutboundEmailAddressProxy(sdkConfig)
+	queueId := d.Id()
+
+	inboundRoute := &platformclientv2.Inboundroute{
+		Id: platformclientv2.String(d.Get("route_id").(string)),
+	}
+
+	emailAddress := platformclientv2.Queueemailaddress{
+		Domain: &platformclientv2.Domainentityref{
+			Id: platformclientv2.String(d.Get("domain_id").(string)),
+		},
+		Route: &inboundRoute,
+	}
+
+	log.Printf("updating outbound email address for queue %s", queueId)
+	_, _, err := proxy.updateRoutingQueueOutboundEmailAddress(ctx, queueId, &emailAddress)
+	if err != nil {
+		return diag.Errorf("failed to update outbound email address for queue %s: %s", queueId, err)
+	}
+	log.Printf("updated outbound email address for queue %s", queueId)
+
+	return readRoutingQueueOutboundEmailAddress(ctx, d, meta)
+}
+
+func deleteRoutingQueueOutboundEmailAddress(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getRoutingQueueOutboundEmailAddressProxy(sdkConfig)
+	queueId := d.Id()
+
+	log.Printf("Removing email address from queue %s", queueId)
+
+	// check if routing queue still exists before trying to remove outbound email address
+	_, resp, err := proxy.getRoutingQueueOutboundEmailAddress(ctx, queueId)
+	if err != nil {
+		if util.IsStatus404(resp) {
```

Comment:
What happens if it is not 404 error code.  Shouldnt we return a diagnostics error

Reply:
You're right we should, I have added the appropriate check